CMakeLists.txt

```diff
@@ -24,3 +27,27 @@ add_compile_options(-Wno-deprecated)  # This is just to avoid some spammy warnin
 find_package(pxr REQUIRED)
 
 add_subdirectory(src)
+
+#-----------------------------------------------------------------------
+# Lint options
+
+# Default treating compiler warnings as errors to OFF, since
+# consumers of this project may use unpredictable toolchains.
+# For dev/CI we should remember to switch this ON, though!
+option(OPENASSETIO_WARNINGS_AS_ERRORS "Treat compiler warnings as errors" OFF)
```

Comment:
[plan] We should namespace these variables (and elsewhere) to be specific to the project, in case someone adds OpenAssetIO as a subproject.

Reply:
Adding specific project naming to the variables : [9b81212](https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/10/commits/9b812126917e4d0d9837bc7e919752aaa60b8cce)